sim/dex.js

```diff
@@ -130,7 +130,7 @@ class ModdedDex {
 		this.abilityCache = new Map();
 		/** @type {Map<string, TypeInfo>} */
 		this.typeCache = new Map();
-		/** @type {Map<string, PureEffect>} */
+		/** @type {Map<string, Effect>} */
```

Comment:
```suggestion
		/** @type {Map<string, Effect | Move>} */
```

There you go; now you can get rid of that other `@ts-ignore`.